modules/bliinkBidAdapter.js

```diff
@@ -75,79 +84,51 @@ export function getKeywords() {
+      return keywords.filter((value) => value).map((value) => value.trim());
     }
   }
 
-  return []
-}
-
-export const parseXML = (content) => {
-  if (typeof content !== 'string' || content.length === 0) return null
-
-  const parser = new DOMParser()
-  let xml;
-
-  try {
-    xml = parser.parseFromString(content, 'text/xml')
-  } catch (e) {}
-
-  if (xml &&
-    xml.getElementsByTagName('VAST')[0] &&
-    xml.getElementsByTagName('VAST')[0].tagName === 'VAST') {
-    return xml
-  }
-
-  return null
+  return [];
 }
 
 /**
  * @param bidRequest
- * @param bliinkCreative
- * @return {{cpm, netRevenue: boolean, requestId, width: (*|number), currency, ttl: number, creativeId, height: (*|number)} & {mediaType: string, vastXml}}
+ * @return {({cpm, netRevenue: boolean, requestId, width: number, currency, ttl: number, creativeId, height: number}&{mediaType: string, vastXml})|null}
  */
-export const buildBid = (bidRequest, bliinkCreative) => {
-  if (!bidRequest && !bliinkCreative) return null
-
-  const body = {
-    requestId: bidRequest.bidId,
-    currency: bliinkCreative.currency,
-    cpm: bliinkCreative.price,
-    creativeId: bliinkCreative.creativeId,
-    width: (bidRequest.sizes && bidRequest.sizes[0][0]) || 1,
-    height: (bidRequest.sizes && bidRequest.sizes[0][1]) || 1,
-    netRevenue: false,
-    ttl: 3600,
-  }
-
-  // eslint-disable-next-line no-mixed-operators
-  if ((bliinkCreative) && bidRequest &&
-    // eslint-disable-next-line no-mixed-operators
-    !bidRequest.bidId ||
-    !bidRequest.sizes ||
-    !bidRequest.params ||
-    !(bidRequest.params.placement)
-  ) return null
-
-  delete bidRequest['bids']
+export const buildBid = (bidRequest) => {
+  if (!bidRequest || !bidRequest.creative || !bidRequest.creative.media_type) return null;
 
-  switch (bliinkCreative.media_type) {
+  let bid;
+  switch (bidRequest.creative.media_type) {
     case VIDEO:
-      return Object.assign(body, {
-        mediaType: VIDEO,
-        vastXml: bliinkCreative.content,
-      })
+      const vastXml = bidRequest.creative.video.content
+      bid = {
+        vastXml,
+        mediaType: 'video',
+        vastUrl: 'data:text/xml;charset=utf-8;base64,' + btoa(vastXml.replace(/\\"/g, '"'))
+      };
+      break;
     case BANNER:
-      return Object.assign(body, {
-        mediaType: BANNER,
-        ad: (bliinkCreative && bliinkCreative.content && bliinkCreative.content.creative && bliinkCreative.content.creative.adm) || '',
-      })
-    default:
+      bid = {
+        ad: bidRequest.creative.banner.adm,
+        mediaType: 'banner',
+      };
       break;
+    default:
+      return null;
   }
-}
+
+  return Object.assign(bid, {
+    cpm: bidRequest.price,
+    currency: bidRequest.currency || 'EUR',
+    creativeId: bidRequest.extras.deal_id,
+    requestId: bidRequest.extras.transaction_id,
+    width: bidRequest.creative[bid.mediaType].width || 1,
+    height: bidRequest.creative[bid.mediaType].height || 1,
```

Comment:
lines 124-127 should use deepAccess to retrieve values to avoid any potential uncaught type errors if the nested properties do not exist in your bid responses